plugins/Wallet/js/locking.js

```diff
@@ -59,7 +53,9 @@ function updateStatus(result) {
 
 // Make wallet api call
 function getStatus() {
-	Siad.apiCall('/wallet', updateStatus);
+	// TODO: Don't understand why the setImmediate is needed, but without it,
+	// the '/wallet' call seems to not return right after a locking/unlocking
```

Comment:
I don't know exactly what you mean by `stacking events` but perhaps. It's not being double-called as far as I know.